server/eclipse-project/src/main/java/au/org/democracydevelopers/corla/endpoint/GetAssertions.java

```diff
@@ -21,5 +21,258 @@
+    /**
+     * RAIRE service endpoint name.
+     */
+    private static final String RAIRE_ENDPOINT = "/raire/get-assertions";
+
+    /**
+     * RAIRE service suffix for csv.
+     */
+    private static final String CSV_SUFFIX = "csv";
+
+    /**
+     * RAIRE service suffix for json.
+     */
+    private static final String JSON_SUFFIX = "json";
+
+    /**
+     * String for "format" query parameter (value will be csv or json).
+     */
+    private static final String FORMAT_PARAM = "format";
+
+    /**
+     * The httpClient used for making requests to the raire-service.
+     */
+    CloseableHttpClient httpClient = HttpClients.createDefault();
+
+
+    /**
+     * The event to return for this endpoint.
+     */
+    private final ThreadLocal<ASMEvent> my_event = new ThreadLocal<ASMEvent>();
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public EndpointType endpointType() {
+        return EndpointType.GET;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String endpointName() {
+        return "/get-assertions";
+    }
+
+    /**
+     * @return No authorization is necessary for this endpoint.
+     */
+    public AuthorizationType requiredAuthorization() { return AuthorizationType.STATE; }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected ASMEvent endpointEvent() {
+        return my_event.get();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void reset() {
+        my_event.set(null);
+    }
+
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String endpointBody(final Request the_request, final Response the_response) {
+        final String prefix = "[endpointBody]";
+
+        final String raireUrl = Main.properties().getProperty(RAIRE_URL, "") + RAIRE_ENDPOINT;
+        String suffix;
+
+        // If csv was requested in the query parameter, hit the get-assertions-csv endpoint; default to json.
+        String format = the_request.queryParamOrDefault(FORMAT_PARAM, JSON_SUFFIX);
+        if (CSV_SUFFIX.equalsIgnoreCase(format)) {
+            suffix = CSV_SUFFIX;
+        } else {
+            suffix = JSON_SUFFIX;
+        }
+
+        // Use the DoS Dashboard to get the risk limit; default to 0 if none is specified.
+        // This is a safe default because the true risk limit cannot be smaller.
+        BigDecimal riskLimit = Persistence.getByID(DoSDashboard.ID, DoSDashboard.class).auditInfo().riskLimit();
+        riskLimit = riskLimit == null ? BigDecimal.ZERO : riskLimit;
+
+        try {
+
+            final ZipOutputStream os = new ZipOutputStream(SparkHelper.getRaw(the_response).getOutputStream());
+            getAssertions(os, riskLimit, raireUrl, suffix);
+
+            the_response.header("Content-Type", "application/zip");
+            the_response.header("Content-Disposition", "attachment; filename*=UTF-8''assertions.zip");
+
+            ok(the_response);
+            return my_endpoint_result.get();
+
```

Comment:
Include e's message in the logging output. 

Should you include contest whose assertions are being requested too?

Reply:
e.getmessage included. I've included contestnames in the ones inside getAssertions, because there is a specific contest at the point those errors are called, whereas here I've left it off because the error is a general thing like bad url config or can't make zip file.